packages/node-opentelemetry/src/gcp.ts

```diff
@@ -0,0 +1,66 @@
+import { SemanticAttributes } from '@opentelemetry/semantic-conventions';
+import { SpanKind, context, trace } from '@opentelemetry/api';
+
+import { initSDK } from './otel';
+import { name as PKG_NAME, version as PKG_VERSION } from '../package.json';
+
+// https://github.com/googleapis/nodejs-pubsub/blob/a4dd4c04e1a508024d133738e4eabbc90eff0ce6/src/subscriber.ts#L787
+const _createSpan = (event) => {
+  const tracer = trace.getTracer(PKG_NAME, PKG_VERSION);
+  let parentSpanContext;
+  let attributes;
+  // extract span context from cloud event
+  if (
+    event.data?.message?.attributes?.['googclient_OpenTelemetrySpanContext']
```

Comment:
should we also check the message type to be from pubsub?

Reply:
added type check